Update katsu to the v3.0 data model part 1

SonQBChau opened the pull request:
## What's new
- Move `primary_site` from `Donor` to `Primary Diagnosis`
- Add `submitter_treatment_id` to `Specimen`
- Move `[“pathological_tumour_staging_system, “pathological_stage_group”,
“pathological_t_category”, “pathological_n_category”, “pathological_m_category”]` from `Specimen` to `Primary Diagnosis`
- Remove `[“lymph_nodes_examined_status”, “lymph_nodes_examined_method”, “number_lymph_nodes_positive”]` from `Primary Diagnosis`
- Remove fields `[“line_of_treatment”, “treatment_setting”]` from `Treatment`
- New `Systemic Therapy` table
- Remove `Chemotherapy, Hormone Therapy, Immunotherapy`
- Remove `submitter_specimen_id` and add `[“surgery_reference_database”, “surgery_reference_identifier”` to `Surgery`
- Remove `[“recurrence_tumour_staging_system”, “recurrence_stage_group”, “recurrence_t_category”, “recurrence_n_category”, “recurrence_m_category”]` from `Follow Up`

### How to test
- No testing, just code review

SonQBChau committed:
Update models.py

- Move “primary_site” from Donor to Primary Diagnosis
- Add “submitter_treatment_id” to Specimen
- Move [“pathological_tumour_staging_system, “pathological_stage_group”,
“pathological_t_category”,  “pathological_n_category”, “pathological_m_category”] from Specimen to Primary Diagnosis
- Remove [“lymph_nodes_examined_status”, “lymph_nodes_examined_method”, “number_lymph_nodes_positive”] from PD
- Remove fields [“line_of_treatment”, “treatment_setting”] from Treatment
- New Systemic Therapy table
- Remove Chemotherapy, Hormone Therapy, Immunotherapy
- Remove “submitter_specimen_id” and add [“surgery_reference_database”, “surgery_reference_identifier” to Surgery
- Remove [“recurrence_tumour_staging_system”, “recurrence_stage_group”, “recurrence_t_category”, “recurrence_n_category”, “recurrence_m_category”] from Follow Up

SonQBChau committed:
Add SystemicTherapy

- Remove Chemotherapy, HormoneTherapy, Immunotherapy

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
Add systemictherapy

SonQBChau committed:
add SystemicTherapy

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
make primary_site optional

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
Add SystemicTherapy

SonQBChau committed:
Update nested_data.py

SonQBChau committed:
Update schemas

SonQBChau committed:
Update schema version

mshadbolt reviewed (approved):
Overall looks good to me, mostly had a look at the models.

kcranston reviewed (approved):
Reviewing against the model documents, this looks good to me.

OrdiNeu commented:
What are the scripts in chord_metadata_service/mohpackets/data/ used for? `data_converter.py` and `data_loader.py` should either be updated or removed

OrdiNeu commented on chord_metadata_service/mohpackets/apis/discovery.py:
```diff
@@ -104,23 +103,8 @@ def discover_sidebar_list(request):
-    # Drugs queryable for hormone therapy
-    hormone_therapy_drug_names = list(
-        HormoneTherapy.objects.exclude(drug_name__isnull=True)
+    systemictherapy_drug_names = list(
```
For readability, should be systemic_therapy
```suggestion
    systemic_therapy_drug_names = list(
```

OrdiNeu commented on chord_metadata_service/mohpackets/apis/discovery.py:
```diff
@@ -129,9 +113,7 @@ def discover_sidebar_list(request):
-        "chemotherapy_drug_names": chemotherapy_drug_names,
-        "immunotherapy_drug_names": immunotherapy_drug_names,
-        "hormone_therapy_drug_names": hormone_therapy_drug_names,
+        "drug_names": systemictherapy_drug_names,
```
```suggestion
        "drug_names": systemic_therapy_drug_names,
```

OrdiNeu commented on chord_metadata_service/mohpackets/apis/ingestion.py:
```diff
@@ -94,11 +90,11 @@ def create_biomarkers(
-def create_chemotherapies(
-    request, payload: List[ChemotherapyIngestSchema], response: HttpResponse
+@router.post("/systemictherapies/")
+def create_systemictherapies(
```
If a similar function is `create_primary_diagnoses`, this should follow that convention.
```suggestion
def create_systemic_therapies(
```

OrdiNeu commented on chord_metadata_service/mohpackets/schemas/filter.py:
```diff
@@ -88,38 +86,24 @@ class TreatmentFilterSchema(FilterSchema):
     actual_cumulative_drug_dose: Optional[int] = Field(None)
+    days_per_cycle: Optional[int] = Field(None)
+    number_of_cycles: Optional[int] = Field(None)
 
```
Missing systemic_therapy_type; it feels like the sort of thing we should be able to filter on.

SonQBChau commented:
> What are the scripts in chord_metadata_service/mohpackets/data/ used for? `data_converter.py` and `data_loader.py` should either be updated or removed

It will be updated after I merge with Marion's synthetic data

SonQBChau committed:
systemic therapy update

- change systemictherapy to systemic_therapy
- add systemic_therapy_type to filter

SonQBChau commented:
change on systemic_therapy as requested

SonQBChau committed:
update schemas

yavyx reviewed (changes requested):
The treatmentTypeEnum still has the old chemo/hormone/immuno values
[See here](https://github.com/CanDIG/katsu/blob/sonchau/model_3_part_1/chord_metadata_service/mohpackets/docs/schema.yml#L4661)

SonQBChau commented:
> The treatmentTypeEnum still has the old chemo/hormone/immuno values [See here](https://github.com/CanDIG/katsu/blob/sonchau/model_3_part_1/chord_metadata_service/mohpackets/docs/schema.yml#L4661)

This change has been addressed in [part 3 ](https://github.com/CanDIG/katsu/pull/248/files#diff-9c4dd410e6ffb078dc81b326de7ed069dbf467816073d1817bcfa92989d79f09L507), so it will be included in the final version

SonQBChau committed:
Update permissible_values.py

SonQBChau committed:
update schemas with enum

SonQBChau commented:
I updated the enum

mshadbolt commented on chord_metadata_service/mohpackets/docs/schema.yml:
```diff
@@ -2554,14 +1879,19 @@ components:
             $ref: '#/components/schemas/NestedSurgerySchema'
           title: Surgeries
           type: array
+        systemictherapies:
```
```suggestion
        systemic_therapies:
```
I think it needs an underscore to be consistent with how we have done it for other schemas

SonQBChau committed:
update schemas systemic_therapies
